### ISSUE TITLE
memory_per_cpu becomes soft_memory_limit

### ISSUE THREAD

**Issue by galerykaeser** [aibasel/lab#97]
While trying to comprehend where in the [environments code](https://github.com/aibasel/lab/blob/cdfdf6f138dc1bb37ec63db1f8a7ab65afeb8155/lab/environments.py) the part is that enables more memory for a task than a single core has access to (like in [this example](https://github.com/aibasel/lab/blob/cdfdf6f138dc1bb37ec63db1f8a7ab65afeb8155/lab/environments.py#L322-L340)), I got confused in [lines 404-406](https://github.com/aibasel/lab/blob/cdfdf6f138dc1bb37ec63db1f8a7ab65afeb8155/lab/environments.py#L404-L406):
The ```soft_memory_limit``` parameter seems to directly depend on ```memory_per_cpu```, which would make it impossible for a task (limited by ```soft_memory_limit```) to get more memory than ```memory_per_cpu```. May this be a bug?

**Comment by jendrikseipp** [comment]
Good catch! I was wondering about this as well recently. I think this code doesn't necessarily lead to a bug because the soft limit can be raised by, for example, the Fast Downward driver. But the code could be adjusted to account for the case when multiple cores are reserved. (Note that memory_per_cpu and cpus_per_task actually mean "cores" instead of "cpus"; Slurm makes loose usage of these two terms.) The problem is that the value of cpus_per_task is currently given as a string in extra_options, so the number of cores is not readily available. What's more is that cpus_per_task is currently only needed for Autonice: it tells Autonice that a single-core task uses more memory than is available per core, so it effectively uses up multiple cores. As far as I'm aware this code has never been used to run multi-core code.

I see the following options:

1. Leave everything as it is and require that users raise the soft limit themselves.
2. Find a better way for Autonice to measure how many cores a user "occupies" and don't recommend to use `--cpus-per-task` anymore.
3. Add a `cpus_per_task` parameter to SlurmEnvironment and compute an accurate soft memory limit.

I think 3. is probably the most robust solution. I'd prefer 2. though because it would keep things simpler. I'll also ask for opinions in the Fast Downward Discord channel.

**Comment by FlorianPommerening** [comment]
I think 3 makes the most sense. I wasn't even aware that lab made the assumption that all runs will be on a single core and if I understand this correctly, a multi-core experiment would currently behave the wrong way.

**Comment by jendrikseipp** [comment]
Sounds good. @galerykaeser do you think you could work on a pull request for this?

**Comment by galerykaeser** [comment]
Yes, I will work on that.
